### ISSUE TITLE
dotnet/installer's release/7.0.1xx branch fails to build with error finding Arcade nuget packages

### ISSUE THREAD

**Issue by omajid** [dotnet/source-build#3021]
### Describe the Bug

An error when trying to build .NET 7's release/7.0.1xx branch from dotnet/installer repo:

```
error : Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1)
```

### Steps to Reproduce

```
$ git clone https://github.com/dotnet/installer
$ cd installer
$ git checkout release/7.0.1xx
Switched to a new branch 'release/7.0.1xx'
Branch 'release/7.0.1xx' set up to track remote branch 'release/7.0.1xx' from 'origin'.
$ git rev-parse HEAD
eaac63b9d78aeffcbbdd76891c5080d05da4f9bb'
$ mkdir -p ../dotnet-tarball
$ ./build.sh /p:Configuration=Release /p:ArcadeBuildTarball=true /p:TarballDir=/home/tester/dotnet-tarball
$ cd ../dotnet-tarball
$ ./prep.sh --bootstrap
$ ./build.sh
Found bootstrap SDK 7.0.100-rc.1.22410.15, bootstrap Arcade 7.0.0-beta.22408.3, bootstrap SourceLink 1.2.0-beta-22407-01
MSBuild version 17.4.0-preview-22403-01+4e58a8330 for .NET
/home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /bl:/home/tester/dotnet-tarball/artifacts/log/Debug/BuildXPlatTasks_0913023939.binlog /fileLoggerParameters:LogFile=/home/tester/dotnet-tarball/artifacts/logs/BuildXPlatTasks_0913023939.log /flp:v=detailed /t:PrepareOfflineLocalTools /home/tester/dotnet-tarball/tools-local/init-build.proj
Build started 9/13/2022 2:39:40 AM.
/home/tester/dotnet-tarball/tools-local/init-build.proj(6,3): warning MSB4011: "/home/tester/dotnet-tarball/git-info/AllRepoVersions.props" cannot be imported again. It was already imported at "/home/tester/dotnet-tarball/Directory.Build.props (219,3)". This is most likely a build authoring error. This subsequent import will be ignored. 
...
```

This fails across multiple opreating systems and architectures:

https://github.com/dotnet/installer at eaac63b9d78aeffcbbdd76891c5080d05da4f9bb on RHEL 9 x64:

```
  /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/RestoreSources.props /p:DotNetBuildOffline=true /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/sourcebuild.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/RestoreSources.props /p:DotNetBuildOffline=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/ /p:ArtifactsDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/ /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/ /p:SourceBuiltBlobFeedDir= /p:EnableSourceControlManagerQueries=false /p:EnableSourceLink=false /p:DeterministicSourcePaths=false /p:DotNetBuildOffline=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1) [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1)
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 ] [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 %5D
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in source-built [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in source-built
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in prebuilt [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in prebuilt
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in reference-packages [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in reference-packages
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found. [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=MSB4236;]The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
      Build failed with exit code 1. Check errors above.
```

Using https://github.com/dotnet/installer at 11f6b8f7129e34d5317f47ccbc110b2b11c05749 on Fedora Rawhide x64:

```
  /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/RestoreSources.props /p:DotNetBuildOffline=true /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/sourcebuild.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/RestoreSources.props /p:DotNetBuildOffline=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/ /p:ArtifactsDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/ /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/ /p:SourceBuiltBlobFeedDir= /p:EnableSourceControlManagerQueries=false /p:EnableSourceLink=false /p:DeterministicSourcePaths=false /p:DotNetBuildOffline=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/x64/Release/PackageVersions.props /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1) [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1)
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 ] [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 %5D
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in source-built [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in source-built
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in prebuilt [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in prebuilt
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in reference-packages [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in reference-packages
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found. [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=MSB4236;]The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
      Build failed with exit code 1. Check errors above.
```

Also seen with https://github.com/dotnet/installer at eaac63b9d78aeffcbbdd76891c5080d05da4f9bb on RHEL 8 arm64:

```
  /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/RestoreSources.props /p:DotNetBuildOffline=true /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/.dotnet/sdk/7.0.100-rc.1.22410.15/MSBuild.dll /nologo -logger:/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/net7.0/Microsoft.DotNet.ArcadeLogging.dll -maxcpucount /m -verbosity:m /v:minimal /v:minimal /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/log/Release/Build.binlog /bl:/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/sourcebuild.binlog /clp:Summary /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:Configuration=Release /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages// /p:Restore=true /p:Build=true /p:ArcadeBuildFromSource=false /p:Rebuild=false /p:Test=false /p:Pack=true /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=true /p:ArcadeBuildFromSource=true /p:CopyWipIntoInnerSourceBuildRepo=true /p:DotNetBuildOffline=true /p:CopySrcInsteadOfClone=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/PackageVersions.props /p:AdditionalSourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/blob-feed/packages/ /p:ReferencePackageNupkgCacheDir=/home/tester/dotnet-tarball/packages/reference/ /p:PreviouslySourceBuiltNupkgCacheDir=/home/tester/dotnet-tarball/packages/previously-source-built/ /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/RestoreSources.props /p:DotNetBuildOffline=true /p:ArcadeInnerBuildFromSource=true /p:DotNetBuildFromSource=true /p:RepoRoot=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/ /p:ArtifactsDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/ /p:ContinuousIntegrationBuild=true /p:SourceBuildOutputDir=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/ /p:SourceBuiltBlobFeedDir= /p:EnableSourceControlManagerQueries=false /p:EnableSourceLink=false /p:DeterministicSourcePaths=false /p:DotNetBuildOffline=true /p:DotNetPackageVersionPropsPath=/home/tester/dotnet-tarball/artifacts/obj/arm64/Release/PackageVersions.props /warnaserror /home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1) [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 7.0.0-beta.22403.1)
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 ] [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 1 version(s) in previously-source-built [ Nearest version: 7.0.0-beta.22408.3 %5D
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in source-built [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in source-built
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in prebuilt [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in prebuilt
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error : - Found 0 version(s) in reference-packages [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=;]- Found 0 version(s) in reference-packages
    /home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found. [/home/tester/dotnet-tarball/packages/restored/ArcadeBootstrapPackage/microsoft.dotnet.arcade.sdk/7.0.0-beta.22408.3/tools/Build.proj]
    ##vso[task.logissue type=error;sourcepath=/home/tester/dotnet-tarball/src/source-build-reference-packages/artifacts/source-build/self/src/artifacts/toolset/restore.proj;linenumber=0;columnnumber=0;code=MSB4236;]The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
      Build failed with exit code 1. Check errors above.
```

**Comment by omajid** [comment]
The previously source-built tarball seems to have a different version?

$ find -iname 'microsoft.dotnet.arcade.sdk*nupkg'
./packages/previously-source-built/Microsoft.DotNet.Arcade.Sdk.7.0.0-beta.22408.3.nupkg

**Comment by MichaelSimons** [comment]
I was able to reproduce this with the rc1 tag.  It appears to only happen when utilizing the bootstrap option with the prep script.

**Comment by crummel** [comment]
Talked to Logan about this and I'm taking a look now.

**Comment by MichaelSimons** [comment]
The source-build's SdkResolver does not appear to be loaded when this happens.  This gets built as part of an msbuild invocation during build.sh before the msbuild invocation to run the actual tarball build.  What I discovered is that after `prep.sh --bootstrap` is run there is a lingering dotnet process.  If you kill this before running build.sh, everything will work.  I have verified this several time and even killing the process after a failed attempt will work.

A question to answer is what is the expected behavior for coping in an SdkResolver?  Is it expected to get picked up or must you kill/restart all dotnet processes?
